### ISSUE TITLE
Support initial value for optional key

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/Defaults#54]
I have:

```swift
extension Defaults.Keys {
	static let refreshInterval = Key<TimeInterval?>("refreshInterval")
}
```

And I would like to specify an initial value, meaning one that is used before any value is set. This is different from `defaultValue`, which is used whenever the value is `nil`, which would not work here as it would make it impossible to set the value to `nil`, which is the whole point of making the value optional.

What I'm thinking:

```swift
extension Defaults.Keys {
	static let refreshInterval = Key<TimeInterval?>("refreshInterval", initialValue: 3600)
}
```

The problem is that there's no good way to differentiate whether a key unset or `nil`.

- `object(forKey:)` returns `nil` whether it's unset or set to `nil`.
- `suite.dictionaryRepresentation().keys.contains(key)` returns `nil` whether it's unset or set to `nil`.
- `suite.persistentDomain(forName:).keys.contains(key)` would work, but we only have the suite instance, not the suite name. And the [suite instance doesn't expose the suite name](https://github.com/feedback-assistant/reports/issues/164)...

Possible solutions:

1. Force passing a `suiteName` parameter too if you want to use `initialValue` with a non-standard suite.
2. Keep track of keys that are set in an internal UserDefaults entry.
3. Other ideas?

I'm leaning towards `2`.

---

// @fredyshox

**Comment by sindresorhus** [comment]
@hank121314 Any opinions on this?

**Comment by hank121314** [comment]
IMO, Although the second solution will be more user friendly, it will create an extra space storage(internal UserDefaults). 

I prefer the first one 😄. 
So bad we could not retrieve the suite name from suite 😭 .

**Comment by sindresorhus** [comment]
Yeah, I agree. The most common use-case is to use the standard suite anyway.

**Comment by sindresorhus** [comment]
Seems like `CFPreferencesCopyValue` does not work either, at least on macOS 13. Maybe it used to work. So it looks like we'll have to manage the state ourselves.

I'm thinking we can store it in a key called `__DEFAULTS__keysThatHaveBeenSet`. It would be an array of keys that have been set once.

The init would be:

```swift
public convenience init<T>(
	_ key: String,
	initialValue: Value,
	suite: UserDefaults = .standard
) where Value == T? {}
```
